gnomad_qc/v4/sample_qc/outlier_filtering.py

```diff
@@ -80,11 +84,77 @@ def get_sample_qc_ht(
     return sample_qc_ht.select_globals()
 
 
-def apply_stratified_filtering_method(
+def apply_filter(
     sample_qc_ht: hl.Table,
     qc_metrics: List[str],
+    filtering_method: str,
+    apply_r_ti_tv_singleton_filter: bool = False,
+    pop_scores_ht: Optional[hl.Table] = None,
     pop_ht: Optional[hl.Table] = None,
     platform_ht: Optional[hl.Table] = None,
+    **kwargs: Any,
+) -> hl.Table:
+    """
+    Apply stratified, regressed, or nearest neighbors outlier filtering method.
+
+    :param sample_qc_ht: Sample QC HT.
+    :param qc_metrics: Specific metrics to use for outlier detection.
+    :param filtering_method: The filtering method to apply to `sample_qc_ht`. One of
+        'stratified', 'regressed', or 'nearest_neighbors'.
+    :param apply_r_ti_tv_singleton_filter: Whether to apply the filtering method to
+        only samples with: Number of singletons (or n_singleton residuals) >
+        median(comparison group number of singletons/n_singleton residuals).
+    :param pop_scores_ht: Optional Table with population PCA scores.
+    :param pop_ht: Optional Table with population assignment.
+    :param platform_ht: Optional Table with platform assignment.
+    :param kwargs: Additional parameters to pass to the requested filtering method
+        function.
+    :return: Table with outlier filter annotations.
+    """
+    # Create dict of expression parameters needed in filtering method.
+    ann_exprs = {}
+    if pop_ht is not None:
+        ann_exprs["pop_expr"] = pop_ht[sample_qc_ht.key].pop
+    if platform_ht is not None:
+        ann_exprs["platform_expr"] = platform_ht[sample_qc_ht.key].qc_platform
+        if filtering_method == "regressed":
+            ann_exprs["platform_scores_expr"] = platform_ht[sample_qc_ht.key].scores
+    if pop_scores_ht is not None:
+        ann_exprs["pop_scores_expr"] = pop_scores_ht[sample_qc_ht.key].scores
+
+    # Run filtering method using defined expressions, and any other passed parameters.
+    if filtering_method == "stratified":
+        ht = apply_stratified_filtering_method(
+            sample_qc_ht, qc_metrics, **ann_exprs, **kwargs
```

Comment:
doesn't apply_stratified_filtering_method only take:
sample_qc_ht
qc_metrics
pop_expr
platform_expr

Reply:
yup, but on the second round the `pop_expr` and the `platform_expr` are in the **kwargs rather than passing in a `pop_ht` and a `platform_ht` to `apply_filter`